src/onCLS.ts

```diff
@@ -62,9 +65,52 @@ export const onCLS = (onReport: CLSReportCallback, opts?: ReportOpts) => {
       let sessionValue = 0;
       let sessionEntries: PerformanceEntry[] = [];
 
+      const initNewCLSMetric = (navigation?: Metric['navigationType']) => {
+        metric = initMetric('CLS', 0, navigation);
+        report = bindReporter(
+          onReport,
+          metric,
+          thresholds,
+          opts!.reportAllChanges
+        );
+        sessionValue = 0;
+        sessionEntries = [];
+      };
+
       // const handleEntries = (entries: Metric['entries']) => {
       const handleEntries = (entries: LayoutShift[]) => {
+        let pageUrl = '';
         entries.forEach((entry) => {
+          if (
```

Comment:
Perhaps a cleaner way to do it is to have the helper class that calls handleEntries automatically split entries based on navigationID?

Then, you dont' need to do this once per entry.

Also, that helper that does the split could call something like "finalizeAndReset", so we wouldn't need to do these tests here at all.  The metric specific handlers would just get a URL and maintain a score...

WDYT?

Reply:
While that may look like cleaner code, I wonder about the performance implications of splitting up the array like versus doing this check everytime? In effect you’d need to do that check anyway for each entry AND copy the entries to a new array. Which sounds like a net negative to me.

WDYT?